node/src/neighborhood/mod.rs

```diff
@@ -3348,6 +4783,161 @@ mod tests {
             Test is written from the standpoint of P. Node q is non-routing.
     */
 
+    #[test]
+    fn find_best_segment_traces_unreachable_country_code_exit_node() {
+        init_test_logging();
+        let mut subject = make_standard_subject();
+        let (recipient, _) = make_node_to_ui_recipient();
+        subject.node_to_ui_recipient_opt = Some(recipient);
+        subject.user_exit_preferences.fallback_preference =
+            FallbackPreference::ExitCountryWithFallback;
+        let message = UiSetExitLocationRequest {
+            fallback_routing: false,
+            exit_locations: vec![CountryCodes {
+                country_codes: vec!["CZ".to_string()],
+                priority: 1,
+            }],
+            show_countries: false,
+        };
+        let db = &mut subject.neighborhood_database;
+        let p = &db.root_mut().public_key().clone();
+        let a = &db.add_node(make_node_record(2345, true)).unwrap();
+        let b = &db.add_node(make_node_record(5678, true)).unwrap();
+        let c = &db.add_node(make_node_record(1234, true)).unwrap();
+        db.add_arbitrary_full_neighbor(p, c);
+        db.add_arbitrary_full_neighbor(c, b);
+        db.add_arbitrary_full_neighbor(c, a);
+        subject.handle_exit_location_message(message, 0, 0);
+
+        let route_cz =
+            subject.find_best_route_segment(p, None, 2, 10000, RouteDirection::Over, None);
+
+        assert_eq!(route_cz, None);
+    }
+
+    #[test]
+    fn route_for_au_country_code_is_constructed_with_fallback_routing() {
+        let mut subject = make_standard_subject();
+        //let db = &mut subject.neighborhood_database;
+        let p = &subject
```

Comment:
proc nazyvas roota, nebo muzes rict subject, pismenem `p`? To nema nejaky hlubsi smysl. Proto bych mozna radeji pouzil treba `r`, at to nemate.